### ISSUE TITLE
ironware spurious \n chars inserted

### ISSUE THREAD

**Issue by nickhilliard** [ytti/oxidized#210]
Ironware appears to insert extra \n characters or not from run to run.  Here's a diff of two runs:

```
@@ -378,6 +378,7 @@
 Compressed Boot-Monitor Image size = 369366, Version:07.3.00T205 (trz07300)
 Code Flash Free Space = 20447232
 SSH@switch#show module
+
 Invalid input -> module
 Type ? for a list
 SSH@switch#
```

Each time this happens, it updates the config repo which triggers post_store update action.  Probably this should be ignored.

**Comment by nickhilliard** [comment]
in all cases that I can see, this extra \n appears to be inserted (or not) immediately after the command is issued to the CLI, e.g.

```
@@ -278,6 +278,7 @@
 end

 SSH@switch#show version
+
   SW: Version 07.4.00hT201 Copyright (c) 1996-2012 Brocade Communications Systems, Inc. All rights reserved.
       Compiled on Feb 24 2015 at 06:35:15 labeled as TIS07400h
       (5764588 bytes) from Primary /tftpboot/TIS07400h.bin
```

**Comment by ytti** [comment]
Perhaps in :all remove all \n in top of the array?

On 3 October 2015 at 11:35, Nick Hilliard <notifications@github.com> wrote:

> in all cases that I can see, this extra \n appears to be inserted (or not)
> immediately after the command is issued to the CLI, e.g.
>
> @@ -278,6 +278,7 @@
>  end
>
>  SSH@switch#show version
> +
>    SW: Version 07.4.00hT201 Copyright (c) 1996-2012 Brocade Communications Systems, Inc. All rights reserved.
>        Compiled on Feb 24 2015 at 06:35:15 labeled as TIS07400h
>        (5764588 bytes) from Primary /tftpboot/TIS07400h.bin
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ytti/oxidized/issues/210#issuecomment-145263419>.
>



-- 
  ++ytti

**Comment by nickhilliard** [comment]
that would probably work. I had a look through another couple of commits; sometimes it adds in "\n " and sometimes "\n".  So perhaps remove all lines at the top of the array which contain only white space chars?

**Comment by ytti** [comment]
That sounds reasonable.

[ytti@ytti.fi ~]% pry
[1] pry(main)> ["\n", " \n", "\n  ", "poop\n", "baz"]
=> ["\n", " \n", "\n  ", "poop\n", "baz"]
[2] pry(main)> ["\n", " \n", "\n  ", "poop\n", "baz"].drop_while { |e|
e.match /^\s+$/ }
=> ["poop\n", "baz"]
[3] pry(main)>

On 3 October 2015 at 13:21, Nick Hilliard <notifications@github.com> wrote:

> that would probably work. I had a look through another couple of commits;
> sometimes it adds in "\n " and sometimes "\n". So perhaps remove all lines
> at the top of the array which contain only white space chars?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ytti/oxidized/issues/210#issuecomment-145274328>.
>



-- 
  ++ytti

**Comment by ytti** [comment]
Is this still relevant?

**Comment by nickhilliard** [comment]
this was fixed in cf591a9
